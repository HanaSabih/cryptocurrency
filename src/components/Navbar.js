import React from 'react';
import { NavLink } from 'react-router-dom';
import Nav from 'react-bootstrap/Nav';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGear, faChevronLeft, faMicrophone } from '@fortawesome/free-solid-svg-icons';

const Navbar = () => (

  <header>

    <Nav className="d-flex justify-content-between align-items-center px-2 py-3 text-white">
      <Nav.Item>
        <NavLink to="/" className=" text-decoration-none fw-bolder fs-5">
          <FontAwesomeIcon icon={faChevronLeft} size="lg" className="burger" />
          {' '}
          2023
        </NavLink>
      </Nav.Item>
      <Nav.Item>
        <NavLink to="/" className=" text-decoration-none fw-bolder"><h4 className="  fw-bolder mb-0">Cryptocurrency Market</h4></NavLink>
      </Nav.Item>
      <Nav.Item>
        <NavLink to="/" className=" text-decoration-none  pe-3"><FontAwesomeIcon icon={faMicrophone} size="lg" className="config fs-5" /></NavLink>
        <NavLink to="/" className=" text-decoration-none "><FontAwesomeIcon icon={faGear} size="lg" className="config fs-5" /></NavLink>
      </Nav.Item>
    </Nav>

  </header>
);

export default Navbar;
