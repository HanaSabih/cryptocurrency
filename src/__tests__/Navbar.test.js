import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('Navbar component', () => {
  test('renders title "CryptoMarket"', () => {
    render(
      <BrowserRouter><Navbar /></BrowserRouter>,
    );

    const title = screen.getByText('Cryptocurrency Market');
    expect(title).toBeInTheDocument();
  });

  test('component match with the snapshot', () => {
    // Assert
    const tree = renderer.create(<BrowserRouter><Navbar /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
