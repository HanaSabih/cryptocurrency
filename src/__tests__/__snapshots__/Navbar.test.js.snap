// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component component match with the snapshot 1`] = `
<header>
  <div
    className="d-flex justify-content-between align-items-center px-2 py-3 text-white nav"
    onKeyDown={[Function]}
  >
    <div
      className="nav-item"
    >
      <a
        aria-current="page"
        className=" text-decoration-none fw-bolder fs-5 active"
        href="/"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-chevron-left fa-lg burger"
          data-icon="chevron-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l192 192c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 278.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-192 192z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        2023
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        aria-current="page"
        className=" text-decoration-none fw-bolder active"
        href="/"
        onClick={[Function]}
      >
        <h4
          className="  fw-bolder mb-0"
        >
          Cryptocurrency Market
        </h4>
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        aria-current="page"
        className=" text-decoration-none  pe-3 active"
        href="/"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-microphone fa-lg config fs-5"
          data-icon="microphone"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M192 0C139 0 96 43 96 96V256c0 53 43 96 96 96s96-43 96-96V96c0-53-43-96-96-96zM64 216c0-13.3-10.7-24-24-24s-24 10.7-24 24v40c0 89.1 66.2 162.7 152 174.4V464H120c-13.3 0-24 10.7-24 24s10.7 24 24 24h72 72c13.3 0 24-10.7 24-24s-10.7-24-24-24H216V430.4c85.8-11.7 152-85.3 152-174.4V216c0-13.3-10.7-24-24-24s-24 10.7-24 24v40c0 70.7-57.3 128-128 128s-128-57.3-128-128V216z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </a>
      <a
        aria-current="page"
        className=" text-decoration-none  active"
        href="/"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-gear fa-lg config fs-5"
          data-icon="gear"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M495.9 166.6c3.2 8.7 .5 18.4-6.4 24.6l-43.3 39.4c1.1 8.3 1.7 16.8 1.7 25.4s-.6 17.1-1.7 25.4l43.3 39.4c6.9 6.2 9.6 15.9 6.4 24.6c-4.4 11.9-9.7 23.3-15.8 34.3l-4.7 8.1c-6.6 11-14 21.4-22.1 31.2c-5.9 7.2-15.7 9.6-24.5 6.8l-55.7-17.7c-13.4 10.3-28.2 18.9-44 25.4l-12.5 57.1c-2 9.1-9 16.3-18.2 17.8c-13.8 2.3-28 3.5-42.5 3.5s-28.7-1.2-42.5-3.5c-9.2-1.5-16.2-8.7-18.2-17.8l-12.5-57.1c-15.8-6.5-30.6-15.1-44-25.4L83.1 425.9c-8.8 2.8-18.6 .3-24.5-6.8c-8.1-9.8-15.5-20.2-22.1-31.2l-4.7-8.1c-6.1-11-11.4-22.4-15.8-34.3c-3.2-8.7-.5-18.4 6.4-24.6l43.3-39.4C64.6 273.1 64 264.6 64 256s.6-17.1 1.7-25.4L22.4 191.2c-6.9-6.2-9.6-15.9-6.4-24.6c4.4-11.9 9.7-23.3 15.8-34.3l4.7-8.1c6.6-11 14-21.4 22.1-31.2c5.9-7.2 15.7-9.6 24.5-6.8l55.7 17.7c13.4-10.3 28.2-18.9 44-25.4l12.5-57.1c2-9.1 9-16.3 18.2-17.8C227.3 1.2 241.5 0 256 0s28.7 1.2 42.5 3.5c9.2 1.5 16.2 8.7 18.2 17.8l12.5 57.1c15.8 6.5 30.6 15.1 44 25.4l55.7-17.7c8.8-2.8 18.6-.3 24.5 6.8c8.1 9.8 15.5 20.2 22.1 31.2l4.7 8.1c6.1 11 11.4 22.4 15.8 34.3zM256 336c44.2 0 80-35.8 80-80s-35.8-80-80-80s-80 35.8-80 80s35.8 80 80 80z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </a>
    </div>
  </div>
</header>
`;
